### PR TITLE
Patterns Adapter & Facade

### PR DESCRIPTION
**Pull request by AntonTmsh**
4.6. The Adapter and Facade Patterns

### CODE REVIEW

**Review by fojman — CHANGES_REQUESTED** [review]
- does this PR contains Facede as well?
- please check all my comments

File: AdapterFacade/Information.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace Epam.NetMentoring.AdapterFacade
+{
+    class Information<T> : IElements<T>
+    {
+        private readonly List<T> _list = new List<T>();
+
+        public Information(List<T> list)
+        {
+            _list = list ?? throw new ArgumentNullException("Cannot create Information class");
```
Comment (fojman) [inline AdapterFacade/Information.cs]: Exception message text does not correspond to what is actually happening:
- ArgumentNullException(nameof(list));

File: AdapterFacade/InformationAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace Epam.NetMentoring.AdapterFacade
+{
+    public class InformationAdapter<T> : IContainer<T>
+    {
+        private IElements<T> _collection; 
+
+        public InformationAdapter(IElements<T> elements)
+        {
+            _collection = elements ?? throw new ArgumentNullException("Cannot create InformationAdapter");
```
Comment (fojman) [inline AdapterFacade/InformationAdapter.cs]: the same for exception message

File: AdapterFacade/Program.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Epam.NetMentoring.AdapterFacade
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            //IContainer<int> elements = new List<int>();
```
Comment (fojman) [inline AdapterFacade/Program.cs]: Guess you forgot to call your code?

**Review by fojman — CHANGES_REQUESTED** [review]

File: AdapterFacade/Information.cs (hunk 1 above)
Comment (AntonTmsh) [reply AdapterFacade/Information.cs]: ok

File: AdapterFacade/InformationAdapter.cs (hunk 2 above)
Comment (AntonTmsh) [reply AdapterFacade/InformationAdapter.cs]: ok

**Review by fojman — APPROVED** [review]
Looks ok, pls merge

**Pull request merged**